GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -3309,6 +3345,61 @@ private void UpdateButtonStates()
                 : TranslatedStrings.ButtonPush;
         }
 
+        private void UICommands_PostRepositoryChanged(object sender, GitUIEventArgs e)
+        {
+            ThreadHelper.FileAndForget(async () =>
+                {
+                    await this.SwitchToMainThreadAsync();
+                    RescanChanges();
+                });
+        }
+
+        private bool ExecuteCommandStaged(int command)
+        {
+            IScriptsRunner scriptsRunner = UICommands.GetRequiredService<IScriptsRunner>();
+            return scriptsRunner.RunScript(command, owner: this, UICommands, new ScriptOptionsProvider(this, Staged));
+        }
+
+        private bool ExecuteCommandUnstaged(int command)
+        {
+            IScriptsRunner scriptsRunner = UICommands.GetRequiredService<IScriptsRunner>();
+            return scriptsRunner.RunScript(command, owner: this, UICommands, new ScriptOptionsProvider(this, Unstaged));
+        }
+
+        private class ScriptOptionsProvider : IScriptOptionsProvider
+        {
+            private FormCommit _formCommit;
+            private FileStatusList? _fileStatusList;
+
+            public ScriptOptionsProvider(FormCommit formCommit, FileStatusList? fileStatusList)
+            {
+                _formCommit = formCommit;
+                _fileStatusList = fileStatusList;
+            }
+
+            IReadOnlyList<string> IScriptOptionsProvider.Options => new[] { "SelectedFiles", "LineNumber" };
+
+            string IScriptOptionsProvider.GetValue(string option)
+            {
+                switch (option)
+                {
+                    case "SelectedFiles":
+                        IEnumerable<string>? selectedFiles = _fileStatusList?.SelectedItems?.Select(item => _formCommit._fullPathResolver.Resolve(item.Item.Name));
+                        if (selectedFiles is null || !selectedFiles.Any())
+                        {
+                            return null;
+                        }
+
+                        return string.Join(" ", selectedFiles.Select(file => "\"" + (EnvUtils.RunningOnWindows() ? file.Replace("\"", "\"\"") : file.Replace("\\", "\\\\").Replace("\"", "\\\"").Replace("\'", "\\\'")) + "\"").ToArray());
```

Comment:
`.Quote()` is not an option, because quoting for command line is different from generic C-style quoting. The quoting format varies by platform. Implemented as `StringExtensions.QuoteForCommandLine` instead.

After testing, `.ToArray()` was not necessary.